lib/katello/tasks/upgrades/3.11/update_puppet_repos.rake

```diff
@@ -1,6 +1,13 @@
 namespace :katello do
   namespace :upgrades do
     namespace '3.11' do
+      def wait_on_task(task)
+        task = SmartProxy.pulp_master.pulp_api.resources.task.poll(task['task_id'])
+        return if Actions::Pulp::AbstractAsyncTask::FINISHED_STATES.include?(task['state'])
+        sleep 1
+        wait_on_task(task)
```

Comment:
I haven't tested this, but is there a chance this will fall over if the call stack gets too deep? I don't know how many iterations that would take.

Reply:
These actions are really short and this is done during the upgrade process so nothing else should really be going on.  If this was during a sync or something where all the pulp workers are tied up for hours, yes it could happen.  The only way i could see this happening is if the pulp workers are not picking up tasks, which a failure there isn't a bad thing

Reply:
a loop may be more readable, but I don't anticipate this having issues. current code looks OK to me as-is.